Tuya climate compilation error

### The problem

I'm getting the following compilation error when trying to update my ME-81H thermostat:
`src/main.cpp: In function 'void setup()':
src/main.cpp:360:41: error: invalid conversion from 'const char*' to 'uint8_t' {aka 'unsigned char'} [-fpermissive]
  360 |   tuya_tuyaclimate->set_active_state_id("datapoint");
      |                                         ^~~~~~~~~~~
      |                                         |
      |                                         const char*
In file included from src/esphome.h:54,
                 from src/main.cpp:3:
src/esphome/components/tuya/climate/tuya_climate.h:18:36: note:   initializing argument 1 of 'void esphome::tuya::TuyaClimate::set_active_state_id(uint8_t)'
   18 |   void set_active_state_id(uint8_t state_id) { this->active_state_id_ = state_id; }
      |                            ~~~~~~~~^~~~~~~~`



### Which version of ESPHome has the issue?

2024.4.0

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2024.4.3

### What platform are you using?

ESP8266

### Board

_No response_

### Component causing the issue

tuya climate

### Example YAML snippet

```yaml
substitutions:
  device_name: warm-floor

packages:
  device_common: !include .device_common.yaml

esphome:
  name: ${device_name}
  platform: ESP8266
  board: esp01_1m

logger:
  baud_rate: 0

uart:
  rx_pin: GPIO3
  tx_pin: GPIO1
  baud_rate: 9600

tuya:
  time_id: sntp_time
  status_pin:
    number: 14
    inverted: true

climate:
  - platform: tuya
    name: "Warm floor"
    switch_datapoint: 1
    target_temperature_datapoint: 16
    target_temperature_multiplier: 1.0
    current_temperature_datapoint: 24
    current_temperature_multiplier: 0.1
    active_state:
      datapoint: 36
      heating_value: 0

switch:
  - platform: "tuya"
    name: "Warm floor lock"
    switch_datapoint: 40

select:
  - platform: "tuya"
    name: "Warm floor sensor selection"
    enum_datapoint: 43
    options:
      0: Internal
      1: Floor
      2: Both
```


### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

--- a/esphome/components/tuya/climate/__init__.py
+++ b/esphome/components/tuya/climate/__init__.py
@@ -208,7 +208,7 @@ async def to_code(config):
         cg.add(var.set_switch_id(switch_datapoint))
 
     if active_state_config := config.get(CONF_ACTIVE_STATE):
-        cg.add(var.set_active_state_id(CONF_DATAPOINT))
+        cg.add(var.set_active_state_id(active_state_config.get(CONF_DATAPOINT)))
         if (heating_value := active_state_config.get(CONF_HEATING_VALUE)) is not None:
             cg.add(var.set_active_state_heating_value(heating_value))
         if (cooling_value := active_state_config.get(CONF_COOLING_VALUE)) is not None:
@@ -219,14 +219,10 @@ async def to_code(config):
             cg.add(var.set_active_state_fanonly_value(fanonly_value))
     else:
         if heating_state_pin_config := config.get(CONF_HEATING_STATE_PIN):
-            heating_state_pin = await cg.gpio_pin_expression(
-                config(heating_state_pin_config)
-            )
+            heating_state_pin = await cg.gpio_pin_expression(heating_state_pin_config)
             cg.add(var.set_heating_state_pin(heating_state_pin))
         if cooling_state_pin_config := config.get(CONF_COOLING_STATE_PIN):
-            cooling_state_pin = await cg.gpio_pin_expression(
-                config(cooling_state_pin_config)
-            )
+            cooling_state_pin = await cg.gpio_pin_expression(cooling_state_pin_config)
             cg.add(var.set_cooling_state_pin(cooling_state_pin))
 
     if target_temperature_datapoint := config.get(CONF_TARGET_TEMPERATURE_DATAPOINT):
@@ -254,11 +250,11 @@ async def to_code(config):
 
     if preset_config := config.get(CONF_PRESET, {}):
         if eco_config := preset_config.get(CONF_ECO, {}):
-            cg.add(var.set_eco_id(CONF_DATAPOINT))
+            cg.add(var.set_eco_id(eco_config.get(CONF_DATAPOINT)))
             if eco_temperature := eco_config.get(CONF_TEMPERATURE):
                 cg.add(var.set_eco_temperature(eco_temperature))
-        if CONF_SLEEP in preset_config:
-            cg.add(var.set_sleep_id(CONF_DATAPOINT))
+        if sleep_config := preset_config.get(CONF_SLEEP, {}):
+            cg.add(var.set_sleep_id(sleep_config.get(CONF_DATAPOINT)))
 
     if swing_mode_config := config.get(CONF_SWING_MODE):
         if swing_vertical_datapoint := swing_mode_config.get(CONF_VERTICAL_DATAPOINT):
@@ -268,7 +264,7 @@ async def to_code(config):
         ):
             cg.add(var.set_swing_horizontal_id(swing_horizontal_datapoint))
     if fan_mode_config := config.get(CONF_FAN_MODE):
-        cg.add(var.set_fan_speed_id(CONF_DATAPOINT))
+        cg.add(var.set_fan_speed_id(fan_mode_config.get(CONF_DATAPOINT)))
         if (fan_auto_value := fan_mode_config.get(CONF_AUTO_VALUE)) is not None:
             cg.add(var.set_fan_speed_auto_value(fan_auto_value))
         if (fan_low_value := fan_mode_config.get(CONF_LOW_VALUE)) is not None:

